### PR TITLE
mavlink: support LOCAL_FRD and BODY_FRD frames

### PR DESCRIPTION
**Pull request by julianoes**
MAV_FRAME_BODY_NED is deprecated. The same behavior of control in world frame but rotated by yaw can now be achieved with MAV_FRAME_LOCAL_FRD.

MAV_FRAME_BODY_FRD can be used for a true body frame.

Follow up to #17958.

### CODE REVIEW

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -943,8 +951,16 @@ MavlinkReceiver::handle_message_set_position_target_local_ned(mavlink_message_t
...
+				if (target_local_ned.coordinate_frame == MAV_FRAME_BODY_FRD) {
+					const matrix::Vector3f acceleration_setpoint{R * acceleration_body_sp};
+					acceleration_setpoint.copyTo(setpoint.acceleration);
+
+				} else {
+					const float yaw = matrix::Eulerf{R}(2);
+					setpoint.acceleration[0] = cosf(yaw) * acceleration_body_sp(0) - sinf(yaw) * acceleration_body_sp(1);
+					setpoint.acceleration[1] = sinf(yaw) * acceleration_body_sp(0) + cosf(yaw) * acceleration_body_sp(1);
+					setpoint.acceleration[2] = acceleration_body_sp(2);
+				}
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_receiver.cpp:963]: @Jaeyoung-Lim does that make any sense?

**Comment by TSC21** [comment]
>MAV_FRAME_BODY_NED is deprecated. The same behavior of control in world frame but rotated by yaw can now be achieved with MAV_FRAME_LOCAL_FRD.

I don't really agree with this approach, but if this was the decision taken, then I guess that's fine.

**Comment by julianoes** [comment]
> I don't really agree with this approach, but if this was the decision taken, then I guess that's fine.

With which part don't you agree? What would be your suggestion?

**Comment by TSC21** [comment]
> > I don't really agree with this approach, but if this was the decision taken, then I guess that's fine.
> 
> With which part don't you agree? What would be your suggestion?

It's not about your implementation. It's about the fact that we replaced one frame with another and the fact we are mixing inertial and body frames.

**Comment by dagar** [comment]
Can you propose something specific?

I believe the only thing locked down is keeping the legacy implementation of `MAV_FRAME_BODY_NED` for compatibility (and because it's mavlink deprecated and barely used).

For everything else this is an opportunity for a fresh start getting things implemented properly and consistently. http://mavlink.io/en/messages/common.html#MAV_FRAME

**Comment by julianoes** [comment]
Bump: should we discuss this on the dev call tomorrow?

**Comment by julianoes** [comment]
I don't think I care about this right now. I can always pick it up another day.

**Pull request closed without merging**